### PR TITLE
Core/BattlePets: Misc fixes

### PR DESCRIPTION
**Pull request by meji46**
**Changes proposed:**

-  Define BattlePetSpeciesFlags enum class.
- Define BattlePetDbFlags enum class.
- Added check to prevent the pet from being caged if its species has flag BattlePetSpeciesFlags::NotTradable.
- Added check to prevent the pet from being caged if it's in battle pet slots.
- Added check to prevent the pet from being caged if its health is below maximum health.
- Only add pet if the species has flag BattlePetSpeciesFlags::WellKnown.
- Added function to check flag BattlePetSpeciesFlags::LegacyAccountUnique to avoid learning copies of unique pets.
- Implemented CMSG_BATTLE_PET_CLEAR_FANFARE.

**Target branch(es):** 3.3.5/master

- [ ] 3.3.5
- [x] master

**Issues addressed:**
None

**Tests performed:**
Builds, tested in-game

**Known issues and TODO list:** (add/remove lines as needed)
None

### CODE REVIEW

File: src/server/game/DataStores/DBCEnums.h [hunk 1]
```diff
@@ -191,7 +191,29 @@ enum AzeriteTierUnlockSetFlags
     AZERITE_TIER_UNLOCK_SET_FLAG_DEFAULT = 0x1
 };
 
-#define BATTLE_PET_SPECIES_MAX_ID 3159
+#define BATTLE_PET_SPECIES_MAX_ID 3189
```
Comment (Shauren) [inline src/server/game/DataStores/DBCEnums.h:194]: Isnt this a 9.1.5 value?

**Review by Shauren — APPROVED** [review]

**Comment by Shauren** [comment]
Nopch is broken, forward declare BattlePetSpeciesEntry in BattlePetMgr.h

**Comment by meji46** [comment]
Done 👍

File: src/server/game/DataStores/DBCEnums.h (hunk 1 above)
Comment (meji46) [reply src/server/game/DataStores/DBCEnums.h]: Oops, you're right. Restored the 9.1.0 value

**Pull request merged**